src/parser.rs

```diff
@@ -4168,63 +4168,85 @@ pub fn parse_typename(tokens: &[Token], index: &mut usize) -> (ParsedType, Optio
+                                    error = error.or(err);
+                                    break;
+                                }
+                                error = error.or(err);
+                                inner_types.push(inner_type);
+                            }
                         }
                     }
-                }
 
-                if *index >= tokens.len() {
-                    unchecked_type = ParsedType::GenericType(
-                        typename,
-                        inner_types,
-                        Span {
-                            file_id: start.file_id,
-                            start: start.start,
-                            end: tokens[tokens.len() - 1].span.end,
-                        },
-                    )
-                } else {
-                    unchecked_type = ParsedType::GenericType(
-                        typename,
-                        inner_types,
-                        Span {
-                            file_id: start.file_id,
-                            start: start.start,
-                            end: tokens[*index].span.end,
-                        },
-                    )
-                }
+                    if *index >= tokens.len() {
+                        unchecked_type = ParsedType::GenericType(
+                            typename,
+                            inner_types,
+                            Span {
+                                file_id: start.file_id,
+                                start: start.start,
+                                end: tokens[tokens.len() - 1].span.end,
+                            },
+                        )
+                    } else {
+                        unchecked_type = ParsedType::GenericType(
+                            typename,
+                            inner_types,
+                            Span {
+                                file_id: start.file_id,
+                                start: start.start,
+                                end: tokens[*index].span.end,
+                            },
+                        )
+                    }
 
-                if error.is_some() {
-                    *index = previous_index;
+                    if error.is_some() {
+                        *index = previous_index;
+                    }
                 }
-            }
+                TokenContents::ColonColon => {
+                    //Type in namespace
+                    *index += 1;
+                    let (inner_type, err) = parse_typename(tokens, index);
+                    if err.is_some() {
+                        error = error.or(err);
+                        return (unchecked_type, error)
+                    }
+                    if let ParsedType::Name(inner_type_name, _) = inner_type {
+                        unchecked_type = ParsedType::Name(
+                            typename + "::" + &inner_type_name,
```

Comment:
Can we give ParsedType a namespace specifier like we have for calls? That way when we resolve it we can walk the namespace path and find the type

Reply:
Sure, ill try and implement that today :)